solr/solrj/src/java/org/apache/solr/client/solrj/impl/Http2SolrClient.java

```diff
@@ -514,7 +522,7 @@ public NamedList<Object> request(SolrRequest<?> solrRequest, String collection)
     try {
       InputStreamResponseListener listener = new InputStreamReleaseTrackingResponseListener();
       req = sendRequest(makeRequest(solrRequest, url, false), listener);
-      Response response = listener.get(idleTimeoutMillis, TimeUnit.MILLISECONDS);
+      Response response = listener.get(requestTimeoutMillis, TimeUnit.MILLISECONDS);
```

Comment:
was weird to be using the idle (AKA socket) timeout in a place where, obviously, the request timeout is what should be used.